PocketKit/Sources/PocketKit/PremiumUpgrade/Model/ReceiptService.swift

```diff
@@ -66,11 +69,13 @@ class AppStoreReceiptService: NSObject, ReceiptService {
 // MARK: StoreKit 1 delegate
 extension AppStoreReceiptService: SKRequestDelegate {
     func requestDidFinish(_ request: SKRequest) {
-        storeKit1Continuation?.resume(returning: request)
+        storeKit1Continuations.forEach { $0.resume(returning: request) }
+        storeKit1Continuations.removeAll()
```

Comment:
@Gio2018 should we throw these in a syncrounus queue to avoid threading issues?

Reply:
you mean because `storeKitContinuations` could potentially be accessed from different threads?

Reply:
yes

Reply:
et voila` @bassrock !